change-validation/src/main/scala/com/normation/plugins/changevalidation/NotificationService.scala

```diff
@@ -0,0 +1,237 @@
+        , param.subject
+        , body
+      )
+    }
+  }
+
+  val configMailPath = "/opt/rudder/etc/plugins/change-validation.conf"
+  val logger         = NamedZioLogger("plugin.change-validation")
+
+  def sendNotification(step: WorkflowNode, cr: ChangeRequest): IOResult[Unit] = {
+    for {
+      serverConfig <- getSMTPConf(configMailPath)
+      emailConf    <- getStepMailConf(step, configMailPath)
+      // todo: get rudder base url from config here -> should I add `domain.name` in RudderConfig ?
+      //                                            -> get it from an existing param : Where ?
+      //                                            -> Mandatory or not ? In my opinion no
+      params       =  extractChangeRequestInfo("XXXXXX", cr)
+      mf           =  new DefaultMustacheFactory()
+      emailBody    <- getContentFromTemplate(mf, emailConf, params)
+      emailSubject <- getSubjectFromTemplate(mf, emailConf.subject, params)
+      _            <- emailService.sendEmail(serverConfig, emailConf.toEnvelop(emailBody).copy(subject = emailSubject))
+    } yield ()
+  }
+
+  private[this] def getConfig(path: String): IOResult[Config] = {
+    val file           = new File(path)
+    IOResult.effectM {
+      for {
+        configResource <- if (file.exists && file.canRead) {
+                            FileSystemResource(file).succeed
+                          } else {
+                            Inconsistency(s"Configuration file not found: ${file.getPath}").fail
+                          }
+      } yield {
+        ConfigFactory.load(ConfigFactory.parseFile(configResource.file))
+      }
+    }
+  }
+
+  private[this] def getSMTPConf(path: String): IOResult[SMTPConf] = {
+    for {
+      config <- getConfig(path)
+      smtp   <- IOResult.effect(s"An error occurs while parsing SMTP conf in ${path}") {
+                  val hostServer = config.getString("smtp.hostServer")
+                  val port       = config.getInt("smtp.port")
+                  val email      = config.getString("smtp.email")
+                  val login      = {
+                    val l = config.getString("smtp.login")
+                    if (l.isEmpty) None else Some(Username(l))
+                  }
+                  val password   = {
+                    val p = config.getString("smtp.password")
+                    if (p.isEmpty) None else Some(p)
+                  }
+                  SMTPConf(
+                    hostServer
+                    , port
+                    , Email(email)
+                    , login
+                    , password
+                  )
+                }
+    } yield smtp
+  }
+
+  private[this] def getStepMailConf(step: WorkflowNode, path: String): IOResult[EmailConf] = {
+    val s = step match {
+      case Validation => "validation"
+      case Deployment => "deployment"
+      case Cancelled  => "cancelled"
+      case Deployed   => "deployed"
+      case _          => "unsupported"
```

Comment:
In that case, we want to have a error for admin/dev. There should not have other cases, and if evolution in change request plugin happens then it should take care of what message send in that case. 
So: change that to be a `IOResult[String]` with an error for unsopported case with a nice message